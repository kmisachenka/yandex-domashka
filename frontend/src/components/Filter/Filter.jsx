import React from 'react';
import * as PropTypes from 'prop-types';

import FilterItem from '../FilterItem';

import styles from './Filter.module.scss';

export default function Filter(props) {
  const { colors, activeColors, toggleColor } = props;
  return (
    <aside className={styles.filter}>
      <h2 className={styles.title}>Заметки</h2>
      <ul className={styles.list}>
        {colors.map(({ id, color }) => (
          <li key={id} className={styles.item}>
            {activeColors.includes(id) ? (
              <FilterItem checked id={id} color={color} onClick={() => toggleColor(id)} />
            ) : (
              <FilterItem id={id} color={color} onClick={() => toggleColor(id)} />
            )}
          </li>
        ))}
      </ul>
    </aside>
  );
}

Filter.propTypes = {
  colors: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      color: PropTypes.string.isRequired,
    })
  ).isRequired,
  activeColors: PropTypes.arrayOf(PropTypes.number).isRequired,
  toggleColor: PropTypes.func.isRequired,
};
