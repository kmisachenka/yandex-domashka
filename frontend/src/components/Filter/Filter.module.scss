@import '../../media';

.filter {
  align-items: center;
  display: flex;

  padding-bottom: 16px;

  @include tablet {
    padding-bottom: 32px;
    padding-top: 16px;
  }
}

.title {
  @include zero;

  display: none;
  font-family: Helvetica, serif;
  font-size: 36px;
  font-weight: normal;
  line-height: 42px;

  margin-right: 16px;

  @include desktop {
    display: block;
  }
}

.list {
  display: flex;
  list-style: none;
  margin: 0;
  -ms-overflow-style: none;

  overflow-x: auto;
  padding: 3px; /* TODO */

  scrollbar-width: none;
  white-space: nowrap;

  &::-webkit-scrollbar {
    background: transparent;
    width: 0;
  }
}

.item {
  display: inline-block;

  margin-left: 16px;
  position: relative;

  @include desktop {
    &:first-child {
      margin-left: 16px;
    }
  }

  &:first-child {
    margin-left: 0;
  }
}
