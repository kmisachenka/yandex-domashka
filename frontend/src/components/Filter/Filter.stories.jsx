import React from 'react';
import { storiesOf } from '@storybook/react';
import Filter from './Filter';

storiesOf('Фильтр', module).add(
  'фильтр',
  () => (
    <Filter
      activeColors={[1, 3, 4]}
      colors={[
        {
          id: 0,
          color: '#E84747',
        },
        {
          id: 1,
          color: '#F2994A',
        },
        {
          id: 2,
          color: '#F2C94C',
        },
        {
          id: 3,
          color: '#219653',
        },
        {
          id: 4,
          color: '#56CCF2',
        },
        {
          id: 5,
          color: '#2F80ED',
        },
        {
          id: 6,
          color: '#9B51E0',
        },
      ]}
    />
  ),
  { viewport: { defaultViewport: 'iphonex' } }
);
