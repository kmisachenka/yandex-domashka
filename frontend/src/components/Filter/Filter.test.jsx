import React from 'react';
import { shallow } from 'enzyme';

import Filter from './Filter';

const colors = [
  {
    id: 0,
    color: '#E84747',
  },
  {
    id: 1,
    color: '#F2994A',
  },
  {
    id: 2,
    color: '#F2C94C',
  },
];

const toggleColor = jest.fn();

describe('Filter', () => {
  it('should match snapshot', () => {
    const activeColors = [0, 1];
    const wrapper = shallow(
      <Filter
        colors={colors}
        activeColors={activeColors}
        toggleColor={toggleColor}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
