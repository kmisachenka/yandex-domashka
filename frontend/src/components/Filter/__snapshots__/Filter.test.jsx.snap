// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match snapshot 1`] = `
<aside
  className="filter"
>
  <h2
    className="title"
  >
    Заметки
  </h2>
  <ul
    className="list"
  >
    <li
      className="item"
      key="0"
    >
      <FilterItem
        checked={true}
        color="#E84747"
        id={0}
        onClick={[Function]}
      />
    </li>
    <li
      className="item"
      key="1"
    >
      <FilterItem
        checked={true}
        color="#F2994A"
        id={1}
        onClick={[Function]}
      />
    </li>
    <li
      className="item"
      key="2"
    >
      <FilterItem
        checked={false}
        color="#F2C94C"
        id={2}
        onClick={[Function]}
      />
    </li>
  </ul>
</aside>
`;
